### PR TITLE
Dev 46 fix towers

### PR DESCRIPTION
**Pull request by Bryukh**

### CODE REVIEW

File: verification/envs/python_3/battle/commander.py [hunk 1]
```diff
@@ -51,12 +62,12 @@ def subscribe_any_item_in_area(self, center, radius, callback):
...
         return self.subscribe('im_stop', callback, {})
 
-    def subscribe_im_in_enemy_firing_range(self, callback):
-        return self.subscribe('im_in_enemy_firing_range', callback, {})
+    def subscribe_enemy_in_my_firing_range(self, callback):
+        return self.subscribe('enemy_in_my_firing_range', callback)
 
-    def subscribe_enemy_item_in_my_firing_range(self, callback):
-        return self.subscribe('item_in_my_firing_range', callback)
+    def subscribe_the_item_out_my_firing_range(self, item_id, callback):
```
Comment (oduvan) [inline verification/envs/python_3/battle/commander.py:68]: not sure about using "the" in the name of the function

File: initial/python_3 [hunk 2]
```diff
@@ -101,23 +101,43 @@ from battle import commander
...
+        unit_in_firing_range(enemies[0])
+    else:
+        client.subscribe_enemy_in_my_firing_range(unit_in_firing_range)
+
+
+def unit_in_firing_range(data, **kwargs):
+    print(data)
     client.attack_item(data['id'])
+    client.subscribe_the_item_is_dead(data['id'], search_next_target)
+    client.subscribe_the_item_out_my_firing_range(data['id'], search_next_target)
```
Comment (oduvan) [inline initial/python_3:117]: this two events can be united in one "on_wait" or "on_done" that will work in the same way as "on_stand" for units when they are required for a new command

File: initial/python_3 [hunk 3]
```diff
@@ -101,23 +101,43 @@ from battle import commander
...
 def attack_next_enemy(data=None, **kawargs):
     enemy = client.ask_nearest_enemy()
     print('ENEMY', enemy)
     client.attack_item(enemy['id'])
-    result = client.subscribe_death_item(enemy['id'], attack_next_enemy)
-attack_next_enemy()
+    result = client.subscribe_the_item_is_dead(enemy['id'], attack_next_enemy)
+
+
+def move_to_X(data=None):
```
Comment (oduvan) [inline initial/python_3:134]: big X ?

File: verification/envs/python_3/battle/commander.py [hunk 4]
```diff
@@ -51,12 +62,12 @@ def subscribe_any_item_in_area(self, center, radius, callback):
     def subscribe_im_stop(self, callback):
         return self.subscribe('im_stop', callback, {})
 
-    def subscribe_im_in_enemy_firing_range(self, callback):
-        return self.subscribe('im_in_enemy_firing_range', callback, {})
+    def subscribe_enemy_in_my_firing_range(self, callback):
+        return self.subscribe('enemy_in_my_firing_range', callback)
```
Comment (oduvan) [inline verification/envs/python_3/battle/commander.py:66]: why not to use range size as argument to the event

``` Python
return self.subscribe('enemy_in_range', callback, {'range': my['firing_range']})
```

File: initial/python_3 (hunk 3 above)
Comment (Bryukh) [inline initial/python_3:134]: Just for testing, for players it will be changed

File: verification/envs/python_3/battle/commander.py (hunk 4 above)
Comment (Bryukh) [inline verification/envs/python_3/battle/commander.py:66]: We don't have this subscribe. I did only required. Later we will add common

File: verification/envs/python_3/battle/commander.py (hunk 1 above)
Comment (Bryukh) [inline verification/envs/python_3/battle/commander.py:68]: How? I would to accent what it's THE item, not any.

File: initial/python_3 (hunk 2 above)
Comment (Bryukh) [inline initial/python_3:117]: A good idea. We can do this in ticket with "stand" -> "idle"

**Pull request merged**